File: 백준/Silver/14400. 편의점 2/편의점 2.py
import sys
input = sys.stdin.readline

n = int(input())
x, y = [], []

for i in range(n):
    x1, y1 = map(int, input().split())
    x.append(x1)
    y.append(y1)

x.sort()
y.sort()

midX = x[n//2]
midY = y[n//2]

ans = 0
for i in range(n):
    ans += abs(x[i] - midX)
    ans += abs(y[i] - midY)

print(ans)

File: 백준/Silver/14400. 편의점 2/편의점 2.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int[] x = new int[n];
        int[] y = new int[n];

        for (int i = 0; i < n; i++){
            st = new StringTokenizer(br.readLine());
            x[i] = Integer.parseInt(st.nextToken());
            y[i] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(x);
        Arrays.sort(y);

        int midX = x[n/2];
        int midY = y[n/2];

        long ans = 0;
        for(int i = 0; i < n; i++){
            ans += Math.abs(x[i] - midX);
            ans += Math.abs(y[i] - midY);
        }
        System.out.println(ans);
    }
}
